apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SpannerClientPartial.cs

```diff
@@ -64,78 +70,100 @@ public partial class SpannerClientImpl
+        }
 
-        partial void Modify_ExecuteSqlRequest(ref ExecuteSqlRequest request, ref CallSettings settings) =>
+        partial void Modify_ExecuteSqlRequest(ref ExecuteSqlRequest request, ref CallSettings settings)
+        {
             // This operations is routed to leader only if the transaction it uses is of a certain type.
             // We don't have that information here so the leader routing header needs to be applied elsewhere.
             ApplyResourcePrefixHeaderFromSession(ref settings, request.Session);
+            ApplyRequestIdToHeader(ref settings);
+        }
 
         partial void Modify_ExecuteBatchDmlRequest(ref ExecuteBatchDmlRequest request, ref CallSettings settings)
         {
             ApplyResourcePrefixHeaderFromSession(ref settings, request.Session);
             MaybeApplyRouteToLeaderHeader(ref settings);
+            ApplyRequestIdToHeader(ref settings);
         }
 
-        partial void Modify_ReadRequest(ref ReadRequest request, ref CallSettings settings) =>
+        partial void Modify_ReadRequest(ref ReadRequest request, ref CallSettings settings)
+        {
             // This operations is routed to leader only if the transaction it uses is of a certain type.
             // We don't have that information here so the leader routing header needs to be applied elsewhere.
             ApplyResourcePrefixHeaderFromSession(ref settings, request.Session);
+            ApplyRequestIdToHeader(ref settings);
+        }
 
         partial void Modify_BeginTransactionRequest(ref BeginTransactionRequest request, ref CallSettings settings)
         {
             ApplyResourcePrefixHeaderFromSession(ref settings, request.Session);
             MaybeApplyRouteToLeaderHeader(ref settings, request.Options?.ModeCase ?? TransactionOptions.ModeOneofCase.None);
+            ApplyRequestIdToHeader(ref settings);
         }
 
         partial void Modify_CommitRequest(ref CommitRequest request, ref CallSettings settings)
         {
             ApplyResourcePrefixHeaderFromSession(ref settings, request.Session);
             MaybeApplyRouteToLeaderHeader(ref settings);
+            ApplyRequestIdToHeader(ref settings);
         }
 
         partial void Modify_RollbackRequest(ref RollbackRequest request, ref CallSettings settings)
         {
             ApplyResourcePrefixHeaderFromSession(ref settings, request.Session);
             MaybeApplyRouteToLeaderHeader(ref settings);
+            ApplyRequestIdToHeader(ref settings);
         }
 
         partial void Modify_PartitionQueryRequest(ref PartitionQueryRequest request, ref CallSettings settings)
         {
             ApplyResourcePrefixHeaderFromSession(ref settings, request.Session);
             MaybeApplyRouteToLeaderHeader(ref settings);
+            ApplyRequestIdToHeader(ref settings);
         }
 
         partial void Modify_PartitionReadRequest(ref PartitionReadRequest request, ref CallSettings settings)
```

Comment:
Can you double check that we are not missing any of these `partial Modify_...`?

Reply:
That's a question for the BatchWrite folks. I think we can skip adding BatchWrite here at all, while we confirm what we want, and we leave a note on the BatchWrite request/design to add this here.